### PR TITLE
ADAPT-2922: Add FontAwesome leading icon to CTA Link; icon refactor

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Add option to add leading FontAwesome icon to CTA link
- Icon refactor to use new HeroIcon and FaIcon component, including CtaLink, CtaButton, external icon in SbLink, Icon Card FA icon, and various card headline icons. Previously, the CtaLink and CtaButton components were using the react-hero-icon library (this is before TailwindLabs come out with their official HeroIcon React package), this updates them to use the official Hero icon package (by using our new HeroIcon component in this repo).
- Add option to CtaGroup component to display children CTAs in grid of 3 column to create the layout in Figma mock
Screen Shot 2021-07-14 at 1 10 20 AM
- Clean up Section, CtaGroup code and logic

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. https://deploy-preview-76--stanford-alumni.netlify.app/test/yvonne/test-basic-page/
2. Look at the CTA links below the Hero banner over different background colors
3. Check the different breakpoints - they should stack vertically from XS to MD breakpoints.
4. https://deploy-preview-76--stanford-alumni.netlify.app/test/yvonne/test-basic-page-interior-no-hero/
5. Scroll to below Heading Level 6 in the main content WYSIWYG, look at the individual CTA links. Check that the different trailing icon options still work, and the new leading FA icon also works.
6. Scroll down to the red poster, the trailing icon in the CTA button should function as before.
6. Scroll further down, the event card headline icon should function as before.
7. Scroll down to the local footer, the CTA links above the social media icons should function as before.
8. https://deploy-preview-76--stanford-alumni.netlify.app/test/yvonne/test-basic-page-interior-hero/
9. Scroll to the ankle, the icon cards with the FontAwesome icons should function as before, as well as the headline trailing icon.

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/cta/ctaGroup.js [hunk 1]
```diff
@@ -1,16 +1,39 @@
...
+            lg={3}
+            gap
+            className="su-gap-y-xs md:su-gap-y-lg su-justify-items-center"
+          >
+            <CreateBloks blokSection={cta} />
+          </DrGrid>
+        </WidthBox>
+      </SbEditable>
+    );
+  }
```
Comment (sherakama) [inline src/components/cta/ctaGroup.js:28]: Is this related to this PR?

Comment (yvonnetangsu) [reply src/components/cta/ctaGroup.js:28]: Yes, added a display option for CTA Group to display the items in a 3 column grid so it can create the look in Kerri's mock. (it has some differences from our regular grid for cards)

**Review by sherakama — APPROVED** [review]
GTG
